### PR TITLE
Refactor logging backend

### PR DESCRIPTION
**Pull request by gaganahluwalia**
## Description

This PR involves changes done to logging to make it consistent across the backend, and apply the appropriate level of logging alongside changes to the Config.ini file to include ErrorHandler and make it more detailed in case of a Error or above level.

## Changelog

- Made the logging consistent across all the backend files
- Included ErrorHandler and DetailedFormatter in the Config.ini

### CODE REVIEW

File: backend/src/router.py [hunk 1]
```diff
@@ -23,16 +24,12 @@ def build_best_next_step_prompt(task, scratchpad):
 def build_plan(task, scratchpad):
     best_next_step_prompt = build_best_next_step_prompt(task, scratchpad)
 
-    # Call model to choose agent
-    logging.info("#####  ~  Calling LLM for next best step  ~  #####")
-
-    logging.info("Scratchpad so far:")
-    logging.info(scratchpad)
+    logger.info("#####  ~  Calling LLM for next best step  ~  #####")
+    logger.info(f"USER - Scratchpad so far: {scratchpad}")
```
Comment (hsauve-scottlogic) [inline backend/src/router.py:28]: Can I ask what 'USER' represents here and why you're using it at times and not others?

**Review by hsauve-scottlogic — COMMENTED** [review]
Looks great Gagan! Logging looks tidier now, I've just asked a couple of questions, thanks

File: backend/src/router.py (hunk 1 above)
Comment (bheyman-scottlogic) [reply backend/src/router.py:28]: My understanding is that this is to represent what we intend to pass to the frontend. Correct me if i'm wrong @gaganahluwalia

File: backend/src/utils/graph_db_utils.py [hunk 2]
```diff
@@ -12,16 +14,13 @@
 
 
 def test_connection():
-    logging.info("testing database connection...")
     connection_healthy = False
     try:
         driver.verify_connectivity()
-        logging.info("database connection verified")
```
Comment (bheyman-scottlogic) [inline backend/src/utils/graph_db_utils.py:19]: I would argue that we want some sort of logging to confirm the db connection is working

**Review by bheyman-scottlogic — APPROVED** [review]
On top of @hsauve-scottlogic comments, I don't have much to add. Good work!

File: backend/src/router.py (hunk 1 above)
Comment (gaganahluwalia) [reply backend/src/router.py:28]: @bheyman-scottlogic  is right, USER is to represent that we intend to pass this one to Frontend in the future.

File: backend/src/director.py [hunk 3]
```diff
@@ -5,23 +5,21 @@
...
 
-logging = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
 engine = PromptEngine()
 director_prompt = engine.load_prompt("director")
 
 
 def question(question):
-    logging.debug("Received utterance: {question}")
```
[the remark this answers is not in the document]
Comment (gaganahluwalia) [reply backend/src/director.py:15]: @hsauve-scottlogic - This line was removed from the code.

File: backend/src/agents/agent.py [hunk 4]
```diff
@@ -20,20 +21,20 @@ class Agent(ABC):
...
 
-        logging.info("#####  ~  Picking Action from tools:  ~  #####")
-        logging.info(create_all_tools_str(self.tools))
+        logger.debug(f"List of tools: {tool_descriptions}")
 
         response = json.loads(call_model(format_prompt, tools_available))
 
-        logging.info("Tool chosen - choice response:")
-        logging.info(json.dumps(response))
+        logger.info(F"USER - Tool chosen: {json.dumps(response)}")
```
[the remark this answers is not in the document]
Comment (gaganahluwalia) [reply backend/src/agents/agent.py]: Changed now.

File: backend/src/utils/graph_db_utils.py (hunk 2 above)
Comment (gaganahluwalia) [reply backend/src/utils/graph_db_utils.py:19]: If there is a failure, then it should go to the exception where we are logging that exception.
logger.exception(f"Database connection failed: {e}")

**Review by hsauve-scottlogic — APPROVED** [review]

**Review by bheyman-scottlogic — APPROVED** [review]

**Review by hsauve-scottlogic — APPROVED** [review]

**Pull request merged**